Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m

```diff
@@ -214,12 +210,51 @@ - (void)ensureContentConstraints
 {
     [NSLayoutConstraint deactivateConstraints:self.contentContraints];
 
+    const int textViewVInset = 5;
+    const int contentHInset = 6;
+    const int contentHSpacing = 6;
+
+    // We want to grow the text input area to fit its content within reason.
+    const CGFloat kMinTextViewHeight = ceil(self.inputTextView.font.lineHeight
+        + self.inputTextView.textContainerInset.top + self.inputTextView.textContainerInset.bottom
+        + self.inputTextView.contentInset.top + self.inputTextView.contentInset.bottom);
+    const CGFloat kMaxTextViewHeight = 100.f;
+    const CGFloat textViewDesiredHeight = (self.inputTextView.contentSize.height + self.inputTextView.contentInset.top
+        + self.inputTextView.contentInset.bottom);
+    const CGFloat textViewHeight = ceil(MAX(kMinTextViewHeight, MIN(kMaxTextViewHeight, textViewDesiredHeight)));
+    const CGFloat kMinContentHeight = kMinTextViewHeight + textViewVInset * 2;
+
     if (self.attachmentToApprove) {
-        self.contentView.hidden = YES;
-        self.attachmentApprovalView.hidden = NO;
+        OWSAssert(self.attachmentView);
+
+        self.leftButtonWrapper.hidden = YES;
```

Comment:
There are important changes here and below around making sure that subviews not relevant to the current mode are hidden and therefore won't receive input events.